runtime/lib/mirrors.cc

```diff
@@ -835,12 +835,59 @@ DEFINE_NATIVE_ENTRY(Mirrors_makeLocalClassMirror, 1) {
                            Object::null_instance());
 }
 
-
 DEFINE_NATIVE_ENTRY(Mirrors_makeLocalTypeMirror, 1) {
   GET_NON_NULL_NATIVE_ARGUMENT(AbstractType, type, arguments->NativeArgAt(0));
   return CreateTypeMirror(type);
 }
 
+DEFINE_NATIVE_ENTRY(Mirrors_makeLocalTypeMirrorWithTypeArguments, 2) {
+  GET_NON_NULL_NATIVE_ARGUMENT(AbstractType, type, arguments->NativeArgAt(0));
+  GET_NON_NULL_NATIVE_ARGUMENT(Array, args, arguments->NativeArgAt(1));
+
+  const Class& clz = Class::Handle(type.type_class());
```

Comment:
@rmacnak-google Thanks! I found the second option myself, but didn't realize there is one more. :)

I've also spent some time with bounded types and I couldn't really get the test case to fail. I can actually get `FBounded<int>` from `reflectType(FBounded, [int])`. 
I see that `ClassFinalizer::FinalizeType()` performs `CheckBounds` but not sure why it doesn't throw an error for me yet. 

Will look into it more tonight, but if you have any quick hints it would help a lot!